File: uhrwerk-core/src/main/java/io/qimia/uhrwerk/config/representation/Metastore.java
package io.qimia.uhrwerk.config.representation;

public class Metastore {
    private String jdbc_url;
    private String jdbc_driver;
    private String user;
    private String pass;

    public Metastore(String jdbc_url, String jdbc_driver, String user, String pass) {
        this.jdbc_url = jdbc_url;
        this.jdbc_driver = jdbc_driver;
        this.user = user;
        this.pass = pass;
    }

    public String getJdbc_url() {
        return jdbc_url;
    }

    public void setJdbc_url(String jdbc_url) {
        this.jdbc_url = jdbc_url;
    }

    public String getJdbc_driver() {
        return jdbc_driver;
    }

    public void setJdbc_driver(String jdbc_driver) {
        this.jdbc_driver = jdbc_driver;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getPass() {
        return pass;
    }

    public void setPass(String pass) {
        this.pass = pass;
    }
}

File: uhrwerk-engine/src/main/scala/io/qimia/uhrwerk/engine/MetaStore.scala
package io.qimia.uhrwerk.engine

import io.qimia.uhrwerk.common.metastore.config._
import io.qimia.uhrwerk.common.metastore.dependency.TableDependencyService
import io.qimia.uhrwerk.common.metastore.model.{MetastoreModel => MetastoreConnInfo}
import io.qimia.uhrwerk.dao._
import io.qimia.uhrwerk.repo.HikariCPDataSource

import java.sql.DriverManager

object MetaStore {

  /** Build a MetaStore object based on a metastore model object which contains the connection details
    * @param connectionInfo MetaStore model object with connection info
    * @return engine metastore
    */
  def build(connectionInfo: MetastoreConnInfo): MetaStore = {
    Class.forName(connectionInfo.getJdbc_driver)

    HikariCPDataSource.initConfig(
      connectionInfo.getJdbc_url,
      connectionInfo.getUser,
      connectionInfo.getPass
    )

    val tableDao = new TableDAO()
    MetaStore(
      new ConnectionDAO(),
      tableDao,
      new TargetDAO(),
      tableDao,
      new PartitionDAO(),
      new PartitionDependencyDAO()
    )
  }
}

case class MetaStore(
    connectionService: ConnectionService,
    tableService: TableService,
    targetService: TargetService,
    tableDependencyService: TableDependencyService,
    partitionService: PartitionService,
    partitionDependencyService: PartitionDependencyService
)
